src/blocking.rs

```diff
@@ -250,118 +234,60 @@ impl BlockingClient {
         txid: &Txid,
         index: u64,
     ) -> Result<Option<OutputStatus>, Error> {
-        let resp = self
-            .agent
-            .get(&format!("{}/tx/{}/outspend/{}", self.url, txid, index))
-            .call();
-
-        match resp {
-            Ok(resp) => Ok(Some(resp.into_json()?)),
-            Err(ureq::Error::Status(code, resp)) => {
-                if is_status_not_found(code) {
-                    return Ok(None);
-                }
-                Err(Error::HttpResponse {
-                    status: code,
-                    message: resp.into_string()?,
-                })
-            }
-            Err(e) => Err(Error::Ureq(e)),
-        }
+        self.get_opt_response_json(&format!("/tx/{}/outspend/{}", txid, index))
     }
 
     /// Broadcast a [`Transaction`] to Esplora
     pub fn broadcast(&self, transaction: &Transaction) -> Result<(), Error> {
-        let resp = self
-            .agent
-            .post(&format!("{}/tx", self.url))
-            .send_string(&serialize(transaction).to_lower_hex_string());
-
-        match resp {
-            Ok(_) => Ok(()), // We do not return the txid?
-            Err(ureq::Error::Status(code, resp)) => Err(Error::HttpResponse {
-                status: code,
-                message: resp.into_string()?,
-            }),
-            Err(e) => Err(Error::Ureq(e)),
+        let mut request = minreq::post(format!("{}/tx", self.url)).with_body(
```

Comment:
Since we only have one post command, I think it's OK to leave this as a one-off function. If in the future a new post function is needed we can see about creating a shared function then.